WordPress/gutenberg
# Site Editor: Add 'area' selection to convert to template part flow.

## Description

Attempts to resolve https://github.com/WordPress/gutenberg/issues/29222 - by adding selection for 'area' in this modal per design suggestion https://github.com/WordPress/gutenberg/issues/29222#issuecomment-791381586 (update to https://github.com/WordPress/gutenberg/pull/30395#issuecomment-810897861)

## How has this been tested?




To Test:
1. Load the Site Editor (must have a block based theme such as tt1-blocks activated).
2. Select an/some existing blocks in the editor (or create new ones).
3. On the block toolbar, click the ellipsis menu and select the "Convert to Template Part" option.
4. A modal should appear to name the template part and select its area (this area selection is what has been added in this PR).  
5. Create the template part and verify its block variation corresponds to the area selected in step 4.

## Screenshots 

## Types of changes





## Checklist:
